iceoryx_posh/include/iceoryx_posh/internal/log/posh_config_logging.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_POSH_LOG_POSH_CONFIG_LOGGING_HPP
+#define IOX_POSH_LOG_POSH_CONFIG_LOGGING_HPP
+
+/// @todo this might be needed to be public when the logger is used in templates
+
+#include "iceoryx_utils/log/logging_free_function_building_block.hpp"
+
+namespace iox
+{
+struct LoggingComponent
+{
+    static constexpr char Ctx[] = "POSH Config";
+    static constexpr char Description[] = "Log context for the POSH configuration library.";
+};
+
+static constexpr auto LogFatal = iox::log::ffbb::LogFatal<LoggingComponent>;
```

Comment:
this is problematic. We already have this in the iox namespace, so if a class gets both includes, we are in trouble. Do we need an additional logger just for configuration?

Reply:
It's now in a separate target so is not accessible. I can rename this logger.

Reply:
It seems the only options are to either rename the log streams or put them in another namespace.
Both solutions are a bit iffy ... What do you think ?

Reply:
I moved all config classes into namespace `iox::config`.